pkg/vulnsrc/vulnsrc.go

```diff
@@ -137,60 +137,86 @@ type Optimizer interface {
 }
 
 type fullOptimizer struct {
-	dbc db.Operation
+	dbConfig db.VulnOperation
+	dbOp     db.Operation
 }
 
 func (o fullOptimizer) Optimize() error {
-	err := o.dbc.ForEachSeverity(func(tx *bolt.Tx, cveID string, _ types.Severity) error {
-		severity, title, description, references := vulnerability.GetDetail(cveID)
-		vuln := types.Vulnerability{
-			Title:       title,
-			Description: description,
-			Severity:    severity.String(),
-			References:  references,
-		}
-		if err := o.dbc.PutVulnerability(tx, cveID, vuln); err != nil {
-			return xerrors.Errorf("failed to put vulnerability: %w", err)
-		}
-		return nil
+	err := o.dbOp.ForEachSeverity(func(tx *bolt.Tx, cveID string, _ types.Severity) error {
+		return o.fullOptimize(cveID, tx)
 	})
 	if err != nil {
 		return xerrors.Errorf("failed to iterate severity: %w", err)
 	}
 
-	if err := o.dbc.DeleteSeverityBucket(); err != nil {
+	if err := o.dbOp.DeleteSeverityBucket(); err != nil {
 		return xerrors.Errorf("failed to delete severity bucket: %w", err)
 	}
 
-	if err := o.dbc.DeleteVulnerabilityDetailBucket(); err != nil {
+	if err := o.dbOp.DeleteVulnerabilityDetailBucket(); err != nil {
 		return xerrors.Errorf("failed to delete vulnerability detail bucket: %w", err)
 	}
 
 	return nil
 
 }
 
+// TODO: Until we can dependency inject, we need to monkey patch sadly
+var (
+	getDetailFunc = vulnerability.GetDetail
+)
+
+func (o fullOptimizer) fullOptimize(cveID string, tx *bolt.Tx) error {
```

Comment:
nit: I feel it is natural `tx` is a first argument.
```suggestion
func (o fullOptimizer) fullOptimize(tx *bolt.Tx, cveID string) error {
```

Reply:
Roger.